### PR TITLE
Revert "remove provisioner checks (#4254)"

### PR DESCRIPTION
**Pull request by Lldenaurois**
This reverts commit 71e76705fb8495fc73f98c4a30fa5a9c1acbf836.

skip check-dependent-cumulus

### CODE REVIEW

File: runtime/rococo/src/lib.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("rococo"),
 	impl_name: create_runtime_str!("parity-rococo-v2.0"),
 	authoring_version: 0,
-	spec_version: 9130,
+	spec_version: 9134,
```
Comment (drahnr) [inline runtime/rococo/src/lib.rs:109]: Do we want this?

File: roadmap/implementers-guide/src/node/utility/provisioner.md [hunk 2]
```diff
@@ -28,6 +28,8 @@ The dispute inherent is similar to a misbehavior report in that it is an attesta
 
 Dispute resolution is complex and is explained in substantially more detail [here](../../runtime/disputes.md).
 
+> TODO: The provisioner is responsible for selecting remote disputes to replay. Let's figure out the details.
```
Comment (rphmeier) [inline roadmap/implementers-guide/src/node/utility/provisioner.md:31]: This can go

**Review by drahnr — APPROVED** [review]

**Review by drahnr — APPROVED** [review]

**Comment by drahnr** [comment]
bot merge

**Comment by drahnr** [comment]
bot merge

**Pull request merged**